src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -356,13 +357,40 @@ public function compileAutoIncrementStartingValues(Blueprint $blueprint, Fluent
      */
     public function compileRenameColumn(Blueprint $blueprint, Fluent $command, Connection $connection)
     {
-        return $connection->usingNativeSchemaOperations()
-            ? sprintf('alter table %s rename column %s to %s',
+        $version = $connection->getServerVersion();
+
+        if (($connection->isMaria() && version_compare($version, '10.5.2', '<'))
+            || (! $connection->isMaria() && version_compare($version, '8.0.3', '<'))
+        ) {
+            $column = collect($connection->getSchemaBuilder()->getColumns($blueprint->getTable()))
+                ->firstWhere('name', $command->from);
+
+            $modifiers = $this->addModifiers($column['type'], $blueprint, new ColumnDefinition([
+                'change' => true,
+                'type' => match ($column['type_name']) {
+                    'bigint' => 'bigInteger',
+                    'int' => 'integer',
+                    'mediumint' => 'mediumInteger',
+                    'smallint' => 'smallInteger',
+                    'tinyint' => 'tinyInteger',
+                    default => $column['type_name'],
+                },
+                'nullable' => $column['nullable'],
+                'default' => $column['default'],
+                'autoIncrement' => $column['auto_increment'],
+                'collation' => $column['collation'],
+                'comment' => $column['comment'],
+            ]));
+
+            return sprintf('alter table %s change %s %s %s',
                 $this->wrapTable($blueprint),
                 $this->wrap($command->from),
-                $this->wrap($command->to)
-            )
-            : parent::compileRenameColumn($blueprint, $command, $connection);
+                $this->wrap($command->to),
+                $modifiers
+            );
+        }
+
+        return parent::compileRenameColumn($blueprint, $command, $connection);
```

Comment:
Can you explain reason for this entire block and what it does?

Reply:
Sure, MySQL 5.7 and MariaDB < 10.5.2 don't support `alter table foo rename column old_name to new_name` so the workaround is to use the `alter table foo change old_name new_name` syntax (doctrine dbal does the same). But to use this syntax we should include all column modifiers, otherwise they wanna be dropped. That is what this block does, checks the DB version, retrieves current modifiers of column and compiles proper alter syntax to rename the column.